### PR TITLE
BUGFIX: SD-932 Fix attribution position

### PR DESCRIPTION
**Pull request by naibinh**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs
- gdc-analytical-designer: [PR](https://github.com/gooddata/gdc-analytical-designer/pull/3093), [client-demo](https://client-demo.na.intgdc.com:50018/analyze/)
- gdc-dashboards: [PR](https://github.com/gooddata/gdc-dashboards/pull/2796), [client-demo](https://client-demo.na.intgdc.com:50020/dashboards/)

# Related Jira tasks
- SD-932: https://jira.intgdc.com/browse/SD-932

### CODE REVIEW

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7289/ : SUCCESS in 5m 51s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3294/ : SUCCESS in 3s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3827/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3295/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3190/ : SUCCESS in 13m 13s

**Comment by naibinh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1596.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3296/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3191/ : SUCCESS in 19m 44s

File: src/components/core/geoChart/GeoChartRenderer.tsx [hunk 1]
```diff
@@ -281,6 +282,7 @@ export default class GeoChartRenderer extends React.Component<IGeoChartRendererP
         const { chart } = this;
         chart.on("click", DEFAULT_LAYER_NAME, this.handleMapClick);
         chart.on("load", this.setupMap);
+        chart.on("load", this.adjustChartHeight);
```
Comment (ptquang86) [inline src/components/core/geoChart/GeoChartRenderer.tsx:285]: Could we put `adjustChartHeight` into `setupMap`?

Comment (naibinh) [reply src/components/core/geoChart/GeoChartRenderer.tsx:285]: I don't think so, `setupMap` is now having bunch of logic.
Separation is better, IMO.

File: src/components/core/geoChart/GeoChartRenderer.tsx [hunk 2]
```diff
@@ -321,6 +323,24 @@ export default class GeoChartRenderer extends React.Component<IGeoChartRendererP
...
 
+    private adjustChartHeight = () => {
+        const { chart, chartRef } = this;
+        if (!chartRef) {
+            return;
+        }
+
+        const chartHeight: number = chartRef.clientHeight;
+        const parentHeight: number = chartRef.parentElement.clientHeight;
+        const shouldResize: boolean = chartHeight <= MIN_HEIGHT && MIN_HEIGHT <= parentHeight;
```
Comment (lanhq) [inline src/components/core/geoChart/GeoChartRenderer.tsx:334]: Where the MIN_HEIGHT come from?

Comment (naibinh) [reply src/components/core/geoChart/GeoChartRenderer.tsx]: It includes zoom control height (60xp) and its margins (20px for top and bottom respectively)
More [detail](https://github.com/gooddata/gooddata-react-components/pull/1596/files#diff-fa62e9a6886bb2448106a41bbcfa6e89R152)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7294/ : SUCCESS in 2m 48s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3301/ : SUCCESS in 3s

**Comment by ptquang86** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3829/ : SUCCESS in 16m 01s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/9/ : SUCCESS in 18s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7298/ : SUCCESS in 2m 35s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1784/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2620/ : SUCCESS in 15m 57s

**Pull request merged**